track_info.c

```diff
@@ -183,6 +186,52 @@ static int doublecmp0(double a, double b)
 	return (x > 0) - (x < 0);
 }
 
+int strnatcmp(const char * str1, const char * str2)
+{
+	int result;
+	char *strcpy1, *strcpy2;
+
+	strcpy1 = (char*)str1;
+	strcpy2 = (char*)str2;
+
+	while (*strcpy1 != '\0' && isspace(*strcpy1))
+		strcpy1++;
+
+	while (*strcpy2 != '\0' && isspace(*strcpy2))
+		strcpy2++;
+
+	unsigned int pos = 0;
+	int num1 = 0, num2 = 0;
+	size_t strlen1 = strlen(strcpy1);
+	size_t strlen2 = strlen(strcpy2);
+
+	while ((size_t)(strcpy1 - str1 + pos) < strlen1 &&
+	       (size_t)(strcpy2 - str2 + pos) < strlen2 &&
+		!isdigit(*(strcpy1+pos)) && !isdigit(*(strcpy2+pos)))
+		pos++;
+
+	result = strcoll(strcpy1, strcpy2);
+
+	if (pos) {
```

Comment:
`if (pos)` branch will no be taken if the filenames consist entirely of digits. `strcoll` will be incorrectly used for this instead of numeric comparison.